### ISSUE TITLE
mac: only enable use mmove hook when a mouse button is actually pressed

### ISSUE THREAD

**Issue by lunixbochs** [knausj85/knausj_talon#974]
The MMOVE tap/hook here causes Talon to waste CPU when the user moves their mouse without buttons pressed: https://github.com/knausj85/knausj_talon/blob/main/code/mouse.py#L387-L400

You can probably get away with leaving it unregistered by default, and only register it between the drag and release commands.

**Comment by nriley** [comment]
I was able to comment this tap usage entirely and dragging seemed to still work with an eye tracker. I was able to drag both with the left mouse button and with the right mouse button (mostly doesn't do anything different but it does in Excel).

@knausj85, looks like this was your change (https://github.com/knausj85/knausj_talon/commit/e31ac723e6a0e75f0f313e4ffb9f3c07fab68fa9). Any idea what problem it was trying to fix, and if this is still a problem?

**Comment by knausj85** [comment]
This was to fix dragging windows so the window moved with the cursor. I don't believe it does anything on Big Sur or later though.

**Comment by nriley** [comment]
OK, think we can just remove it then, or would it make sense to make it conditional on macOS version? I don't have a 10.15 Mac any more.

**Comment by knausj85** [comment]
#1093 addressed this by removing the tap code

**Comment by brief** [comment]
Re-opening this issue. Drag doesn't work for me on Ventura (13.5 or any previous version) without this code. I can't select text or move windows or drag files. It just single clicks. Notably, I am not using an eye tracker.

I'm still fairly new to Talon and `drag` has never worked for me (in the public or beta), so I always just assumed it required an eye tracker. Mispronounced something in Google Sheets today and it worked, but I couldn't get it to activate anywhere else. Found this issue, tried the code, and it now works everywhere.

Per @lunixbochs's suggestion, I've updated things to register the tap on `mouse_drag()` and unregister things on `mouse_drag_end()`. Happy to submit a pull request if others can replicate the issue and there's no better way of fixing it.
